shell/profiling/sampling_profiler.h

```diff
@@ -0,0 +1,57 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PROFILING_SAMPLING_PROFILER_H_
+#define FLUTTER_SHELL_PROFILING_SAMPLING_PROFILER_H_
+
+#include <functional>
+#include <memory>
+#include <optional>
+
+#include "flutter/fml/task_runner.h"
+#include "flutter/fml/trace_event.h"
+
+namespace flutter {
+
+/**
+ * @brief CPU usage stats. `num_threads` is the number of threads owned by the
+ * process. It is to be noted that this is not per shell, there can be multiple
+ * shells within the process. `total_cpu_usage` is the percentage (between [0,
+ * 100]) cpu usage of the application. This is across all the cores, for example
+ * an application using 100% of all the core will report `total_cpu_usage` as
+ * `100`, if it has 100% across 2 cores and 0% across the other cores, embedder
+ * must report `total_cpu_usage` as `50`.
+ */
+struct CpuUsageInfo {
+  uint32_t num_threads;
+  double total_cpu_usage;
+};
+
+struct ProfileSample {
```

Comment:
We may need doxygen docs for all public classes and functions.

Reply:
done, i've added them for places where i think they would add value and left them out at places where i don't think they'd add more value over the method name for example.

Reply:
BTW, I have found this plugin to be really useful in helping dealing with some of the more tedious aspects of generating documentation stubs https://packagecontrol.io/packages/DoxyDoxygen